identity_did/src/did_url.rs

```diff
@@ -535,6 +534,33 @@ pub(crate) const fn is_char_fragment(ch: char) -> bool {
   is_char_path(ch) || ch == '?'
 }
 
+pub(crate) fn is_valid_percent_encoded_char(s: &str) -> bool {
+  let mut chars = s.chars();
+  let Some('%') = chars.next() else { return false };
+  chars.take(2).all(|c| c.is_ascii_hexdigit() && !c.is_ascii_lowercase())
```

Comment:
I think, we should check the length of `chars.take(2)`, as this would return an iterator with a length of 0 or 1, making wrongly percent encoded elements at the end of an input valid, e.g. `relative_url.set_path(Some("/p%A"))` or `relative_url.set_path(Some("/p%"))`.

> Creates an iterator that yields the first n elements, or fewer if the underlying iterator ends sooner.

(from [docs](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.take))